PurrSoft/PurrSoft.Application/Commands/AnimalCommands/AnimalCommands.cs

```diff
@@ -0,0 +1,36 @@
+﻿using PurrSoft.Application.Common;
+using PurrSoft.Domain.Entities;
+using PurrSoft.Domain.Entities.Enums;
+
+namespace PurrSoft.Application.Commands.AnimalCommands;
+
+public class AnimalGetCommand: BaseRequest<CollectionResponse<Animal>>
+{
+
+}
+
+public class AnimalCreateCommand: BaseRequest<CommandResponse<int>>
+{
+    public AnimalType AnimalType { get; set; }
+    public string? Name { get; set; }
+    public int YearOfBirth { get; set; }
+    public string? Gender { get; set; }
+    public Boolean Sterilized { get; set; }
+    public string? ImageUrl { get; set; }
+}
+
+public class  AnimalUpdateCommand: BaseRequest<CommandResponse<Animal>>
```

Comment:
Also do you really want to send the object? I would say it is better to send a DTO. First of all when you will send the enum to the frontend it will say the number for the  AnimalType. like 1 instead of Cat.

Reply:
I think the animal entity itself is pretty small in size with a small number of attributes that are all needed in the frontend. Is it necessary to add a Dto?